pkgs/dkg/drand.go

```diff
@@ -582,3 +654,258 @@ func (o *LocalOwner) Resign(reqID [24]byte, r *wire.ResignMessage) (*wire.Transp
 		Version:    o.version,
 	}, nil
 }
+
+// InitReshare initiates a resharing owner of dkg protocol
+func (o *LocalOwner) Reshare(reqID [24]byte, reshare *spec.Reshare, commitsPoints []kyber.Point) (*wire.Transport, error) {
+	// sanity check
+	if o.data != nil {
+		return nil, fmt.Errorf("data already exist at local instance: %v", o.data)
+	}
+	o.data = &DKGdata{}
+	var commits []byte
+	for _, point := range commitsPoints {
+		b, _ := point.MarshalBinary()
+		commits = append(commits, b...)
+	}
+	o.data.reshare = reshare
+	o.data.reqID = reqID
+	kyberLogger := o.Logger.With(zap.String("reqid", fmt.Sprintf("%x", o.data.reqID[:])))
+	o.board = board.NewBoard(
+		kyberLogger,
+		func(msg *wire.KyberMessage) error {
+			kyberLogger.Debug("server: broadcasting kyber message")
+			byts, err := msg.MarshalSSZ()
+			if err != nil {
+				return err
+			}
+
+			trsp := &wire.Transport{
+				Type:       wire.ReshareKyberMessageType,
+				Identifier: o.data.reqID,
+				Data:       byts,
+				Version:    o.version,
+			}
+
+			// todo not loop with channels
+			go func(trsp *wire.Transport) {
+				if err := o.Broadcast(trsp); err != nil {
+					o.Logger.Error("broadcasting failed", zap.Error(err))
+				}
+			}(trsp)
+
+			return nil
+		},
+	)
+
+	eciesSK, pk := initsecret(o.Suite)
+	o.data.secret = eciesSK
+	bts, _, err := CreateExchange(pk, commits)
+	if err != nil {
+		return nil, err
+	}
+	return &wire.Transport{
+		Type:       wire.ReshareExchangeMessageType,
+		Identifier: reqID,
+		Data:       bts,
```

Comment:
Why do we use spec.Result as the output data for Resign function above (line 639-655) but not here (and init)?

Reply:
At reshare we have more steps, but at resign only one